core/common/src/main/java/alluxio/AbstractClient.java

```diff
@@ -194,6 +194,13 @@ public synchronized void connect() throws IOException, ConnectionFailedException
       } catch (TTransportException e) {
         LOG.error("Failed to connect (" + retry.getRetryCount() + ") to " + getServiceName() + " "
             + mMode + " @ " + mAddress + " : " + e.getMessage());
+        if (e.getMessage().contains("java.net.SocketTimeoutException")) {
+          // Do not retry if socket timeout.
+          String message = "Thrift transport open times out. Please check whether the "
+              + "authentication types match between client and server. Note that NOSASL client "
+              + "is not able to connect to servers with SIMPLE security mode.";
+          throw new IOException(message, e);
```

Comment:
nit: I would prefer to using a shorter timeout (say 1 min or even shorter) for connection timeout and retry several times if it fails (say 5) with error logging.  We can still keep the 10min timeout for socket io if we want though I feel that is too long (in agreement with your opinion).